### PR TITLE
feat: Support for multiple forms of JSON

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #31051

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/parser/planparserv2/plan_parser_v2_test.go [hunk 1]
```diff
@@ -881,17 +882,17 @@ func Test_InvalidJSONContains(t *testing.T) {
...
 		`json_contains(StringField, "a")`,
 		`json_contains(A, StringField > 5)`,
 		`json_contains(A)`,
 		`json_contains(A, 5, C)`,
-		`json_contains(JSONField, 5)`,
-		`json_Contains(JSONField, 5)`,
-		`JSON_contains(JSONField, 5)`,
+		//`json_contains(JSONField, 5)`,
+		//`json_Contains(JSONField, 5)`,
+		//`JSON_contains(JSONField, 5)`,
```
Comment (czs007) [inline internal/parser/planparserv2/plan_parser_v2_test.go:895]: move to normal cases

File: internal/parser/planparserv2/parser_visitor.go [hunk 2]
```diff
@@ -36,9 +36,9 @@ func (v *ParserVisitor) translateIdentifier(identifier string) (*ExprWithType, e
 	if identifier != field.Name {
 		nestedPath = append(nestedPath, identifier)
 	}
-	if typeutil.IsJSONType(field.DataType) && len(nestedPath) == 0 {
-		return nil, fmt.Errorf("can not comparisons jsonField directly")
-	}
+	//if typeutil.IsJSONType(field.DataType) && len(nestedPath) == 0 {
+	//	return nil, fmt.Errorf("can not comparisons jsonField directly")
```
Comment (czs007) [inline internal/parser/planparserv2/parser_visitor.go:40]: just delete this

**Pull request merged**